### PR TITLE
Untested firestore fix for GetPrivateCollection

### PR DESCRIPTION
**Pull request by cdiddy77**
I did test that it creates an ArtMeta, in an isolated project, but I don't really know how to run and debug the plugin. I imagine it isn't more than a handful of fixes away from a working version.

### CODE REVIEW

File: Editor/DevPanel.cs [hunk 1]
```diff
@@ -611,18 +611,60 @@ public class QueueResponse {
...
+            if (fields.Type != JTokenType.Object) {
+                return output;
+            }
+
+            foreach (var artMetaJson in fields.Children<JProperty>()) {
+                string name = artMetaJson.Name;
+                ArtMeta value = new ArtMeta();
+                value.artId = name;
+                foreach (var field in artMetaJson.Value["mapValue"]["fields"].Children()) {
+                    if (field.Value<JProperty>().Name == "artist") {
```
Comment (chornesays) [inline Editor/DevPanel.cs:644]: Great candidate for a switch vs a giant if:else

File: Editor/DevPanel.cs [hunk 2]
```diff
@@ -611,18 +611,60 @@ public class QueueResponse {
...
+                        value.artist = previewObject.Value<string>("stringValue");
+                    } else if (field.Value<JProperty>().Name == "creationTime") {
+                        var previewObject = field.Value<JProperty>().Value as JObject;
+                        value.creationTime = previewObject.Value<string>("integerValue");
+                    } else if (field.Value<JProperty>().Name == "title") {
+                        var previewObject = field.Value<JProperty>().Value as JObject;
+                        value.title = previewObject.Value<string>("stringValue");
+                    } else if (field.Value<JProperty>().Name == "version") {
+                        var previewObject = field.Value<JProperty>().Value as JObject;
+                        value.version = previewObject.Value<string>("integerValue");
```
Comment (chornesays) [inline Editor/DevPanel.cs:655]: version isnt a field in ArtMeta in the plugin. Everything else was good.

**Review by chornesays — APPROVED** [review]
THANK YOU FOR HANDLING THIS. Parsing JSON in C# is not fun

File: Editor/DevPanel.cs (hunk 1 above)
Comment (cdiddy77) [reply Editor/DevPanel.cs]: Done

File: Editor/DevPanel.cs (hunk 2 above)
Comment (cdiddy77) [reply Editor/DevPanel.cs]: Fixed. It caused a wee crasheebooboo

**Comment by cdiddy77** [comment]
I debugged it and it is working AFAICT

**Pull request merged**